libs/@local/hash-graph-types/rust/src/knowledge/entity/provenance.rs

```diff
@@ -7,9 +7,135 @@ use bytes::BytesMut;
+pub struct Tool {
+    /// The name of the tool.
+    pub name: String,
+
+    /// The tool version, in whatever format the tool natively provides.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub version: Option<String>,
+
+    /// The tool version in the format specified by Semantic Versioning 2.0.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = String))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub semantic_version: Option<semver::Version>,
+
+    /// The organization or company that produced the tool.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub organization: Option<String>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub enum SourceType {
+    Webpage,
+    Document,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct SourceProvenance {
+    #[serde(rename = "type")]
+    pub ty: SourceType,
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub authors: Vec<String>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub actor_type: Option<ActorType>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = Timestamp))]
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "codec::serde::time::option"
+    )]
+    pub first_published: Option<OffsetDateTime>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = Timestamp))]
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "codec::serde::time::option"
+    )]
+    pub last_updated: Option<OffsetDateTime>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = Timestamp))]
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "codec::serde::time::option"
+    )]
+    pub loaded_at: Option<OffsetDateTime>,
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = Timestamp))]
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "codec::serde::time::option"
+    )]
+    pub analyzed_at: Option<OffsetDateTime>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "kebab-case")]
+pub enum OriginType {
+    WebApp,
+    MobileApp,
+    BrowserExtension,
+    Api,
+    Flows,
```

Comment:
Should this be singular?
```suggestion
    Flow,
```